### PR TITLE
improvements to inmem/Measurement.TagSets API

### PR DESCRIPTION
**Pull request by stuartcarnie**
```
benchmark                                             old ns/op     new ns/op     delta
BenchmarkMeasurement_TagSetsNoDimensions_1000-8       234054        117315        -49.88%
BenchmarkMeasurement_TagSetsDimensions_1000-8         996838        313313        -68.57%
BenchmarkMeasurement_TagSetsNoDimensions_100000-8     58940464      39452117      -33.06%
BenchmarkMeasurement_TagSetsDimensions_100000-8       175612060     70195562      -60.03%

benchmark                                             old allocs     new allocs     delta
BenchmarkMeasurement_TagSetsNoDimensions_1000-8       1026           26             -97.47%
BenchmarkMeasurement_TagSetsDimensions_1000-8         8026           2029           -74.72%
BenchmarkMeasurement_TagSetsNoDimensions_100000-8     100064         64             -99.94%
BenchmarkMeasurement_TagSetsDimensions_100000-8       800064         200067         -74.99%

benchmark                                             old bytes     new bytes     delta
BenchmarkMeasurement_TagSetsNoDimensions_1000-8       117080        69080         -41.00%
BenchmarkMeasurement_TagSetsDimensions_1000-8         549081        117176        -78.66%
BenchmarkMeasurement_TagSetsNoDimensions_100000-8     23298264      18498265      -20.60%
BenchmarkMeasurement_TagSetsDimensions_100000-8       66498276      23298360      -64.96%
```

###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass

### CODE REVIEW

**Comment by stuartcarnie** [comment]
Thanks @corylanou

File: tsdb/index/inmem/meta.go [hunk 1]
```diff
@@ -308,6 +308,13 @@ func (m *Measurement) TagSets(shardID uint64, opt influxql.IteratorOptions) ([]*
 		return nil, err
 	}
 
+	var dims []string
+	if len(opt.Dimensions) > 0 {
+		dims = make([]string, len(opt.Dimensions))
+		copy(dims, opt.Dimensions)
+		sort.Strings(dims)
```
Comment (jwilder) [inline tsdb/index/inmem/meta.go]: Can this sort be done after the lock is released?

Comment (stuartcarnie) [inline tsdb/index/inmem/meta.go]: There is no lock established until line 318, after this block

Comment (jwilder) [inline tsdb/index/inmem/meta.go]: Ah.  I misread L318.  Never mind. :)

**Pull request merged**